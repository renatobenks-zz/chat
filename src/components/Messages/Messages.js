// @flow

import * as React from 'react';
import styled from 'styled-components';

type Props = {
  children: React.Element<any>,
};

const Messages = styled.div.attrs({
  children: ({ children }: Props) => children,
})`
  display: flex;
  flex-flow: column wrap;
  flex: 1;
  justify-content: stretch;
  align-items: flex-start;
  padding: 1rem;
`;

export default Messages;
