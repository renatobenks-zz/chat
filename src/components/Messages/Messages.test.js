import React from 'react';

import { render } from '../../../test/helpers';

import Messages from './Messages';

const renderer = component => render(component, { withTheme: true });

it('messages with children', () => {
  expect(
    renderer(
      <Messages>
        <span>message</span>
      </Messages>,
    ),
  ).toMatchSnapshot();
});

it('messages with no children', () => {
  expect(renderer(<Messages />)).toMatchSnapshot();
});
