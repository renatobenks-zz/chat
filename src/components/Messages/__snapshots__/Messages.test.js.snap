// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`messages with children 1`] = `
<div
  className="sc-bdVaJa gAeFo"
>
  <span>
    message
  </span>
</div>
`;

exports[`messages with no children 1`] = `
<div
  className="sc-bdVaJa gAeFo"
/>
`;
